src/snapred/backend/service/ReductionService.py

```diff
@@ -424,3 +461,29 @@ def _groupByVanadiumVersion(self, requests: List[SNAPRequest]):
     def getCompatibleMasks(self, request: ReductionRequest) -> List[WorkspaceName]:
         runNumber, useLiteMode = request.runNumber, request.useLiteMode
         return self.dataFactoryService.getCompatibleReductionMasks(runNumber, useLiteMode)
+
+    def artificialNormalization(self, request: CreateArtificialNormalizationRequest):
+        ingredients = ArtificialNormalizationIngredients(
+            peakWindowClippingSize=request.peakWindowClippingSize,
+            smoothingParameter=request.smoothingParameter,
+            decreaseParameter=request.decreaseParameter,
+            lss=request.lss,
+        )
+        artificialNormWorkspace = ArtificialNormalizationRecipe().executeRecipe(
+            InputWorkspace=request.diffractionWorkspace,
+            Ingredients=ingredients,
+            OutputWorkspace=request.outputWorkspace,
+        )
+        return artificialNormWorkspace
+
+    def grabDiffractionWorkspaceforArtificialNorm(self, request: ReductionRequest):
+        calVersion = None
+        calVersion = self.dataFactoryService.getThisOrLatestCalibrationVersion(request.runNumber, request.useLiteMode)
+        calRecord = self.dataFactoryService.getCalibrationRecord(request.runNumber, request.useLiteMode, calVersion)
+        filePath = self.dataFactoryService.getCalibrationDataPath(request.runNumber, request.useLiteMode, calVersion)
+        diffCalOutput = calRecord.workspaces[wngt.DIFFCAL_OUTPUT][0]
+        diffcalOutputFilePath = str(filePath) + "/" + str(diffCalOutput) + ".nxs.h5"
+
+        groceries = self.groceryService.fetchWorkspace(diffcalOutputFilePath, "diffractionWorkspace")
```

Comment:
Could you move this logic to a method on groceryService instead? I can imagine it would be very useful elsewhere to be able to read a calibration's run data with a simple (runnumber, litemode, calversion).

It would also be more in line with the rest of the architecture.